src/analysis/inner_product_space/calculus.lean

```diff
@@ -251,6 +251,15 @@ lemma differentiable_on.dist (hf : differentiable_on ℝ f s) (hg : differentiab
   differentiable_on ℝ (λ y, dist (f y) (g y)) s :=
 λ x hx, (hf x hx).dist (hg x hx) (hne x hx)
 
+lemma has_fderiv_at_norm_sq : has_fderiv_at (λ (x : E), ∥x∥ ^ 2) (0 : E →L[ℝ] ℝ) 0 :=
```

Comment:
This should be true in any normed space. Basically, you want to prove that `∥x∥ ^ 2 = o(∥x∥)` as `x` tends to zero.

Reply:
Quite true!  I solved this in the short term by removing this lemma (I realised I didn't even need it).